libp2p/NodeTable.cpp

```diff
@@ -21,8 +21,16 @@
 
 #include "NodeTable.h"
 using namespace std;
-using namespace dev;
-using namespace dev::p2p;
+
+namespace dev
+{
+namespace p2p
+{
+inline bool operator==(
```

Comment:
(this is needed to find a `weak_ptr` corresponding to a `shared_ptr`)
The whole idea of storing `shared_ptr<NodeEntry>` in a separate map and then storing corresponding `weak_ptr`s in the actual buckets of the Node Table seems unnecessary complicated to me, I will file a separate issue about this.